compiler/src/main/scala/dx/compiler/ApplicationCompiler.scala

```diff
@@ -652,12 +652,13 @@ case class ApplicationCompiler(typeAliases: Map[String, Type],
         "hidden" -> JsBoolean(hidden)
     )
     // look for ignoreReuse in runtime hints and in extras - the later overrides the former
-    val ignoreReuse = applet.requirements
-      .collectFirst {
-        case IgnoreReuseRequirement(value) => value
-      }
+    val ignoreReuse = extras
+      .flatMap(_.ignoreReuse)
       .orElse(
-          extras.flatMap(_.ignoreReuse)
+          applet.requirements
+            .collectFirst {
+              case IgnoreReuseRequirement(value) => value
+            }
```

Comment:
what's provided in extras should override what is given in the task runtime.

The workflowCompiler does not have this issue, because it does not have its own runtime section therefore the only ignoreReuse comes from extras.json